File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> triangle = { {1} };
        for (int i = 1; i < numRows; ++i) {
            vector<int> row = { 1 };
            int lastRowIdx = triangle.size() - 1;
            for (int j = 1; j <= lastRowIdx; ++j) {
                row.push_back(triangle[lastRowIdx][j-1] + triangle[lastRowIdx][j]);
            };
            row.push_back(1);
            triangle.push_back(row);
        };
        return triangle;
    }; 
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = [[1]]

        if numRows == 1:
            return triangle

        numRows -= 1
        while numRows:
            row = []
            i = 0
            while i <= len(triangle[-1]):
                j = i - 1
                left = 0 if j < 0 else triangle[-1][j]
                right = 0 if i == len(triangle[-1]) else triangle[-1][i]
                row.append(left + right)
                i += 1
            triangle.append(row)
            numRows -= 1

        return triangle